enhancements/olm-feature-gate.md

```diff
@@ -0,0 +1,395 @@
+
+~~~
+spec:
+  type: object
+    properties:
+      features:
+        description: Features contains the list of configurable OLM features.
+        type: object
+        properties:
+          disableCopiedCSVs:
+            description: DisableCopiedCSVs is used to disable OLM's "Copied CSV" feature for operators installed at the cluster scope, where a cluster scoped operator is one that has been installed in an OperatorGroup that targets all namespaces. When reenabled, OLM will recreate the "Copied CSVs" for each cluster scoped operator.
+            type: boolean
+          gates:
+            description: gates are used to activate/deactivate features.
+            type: array
+            items:
+              description: A gate is a name/value pair that contains whether a specific feature is enabled or not.
+              type: object
+              required:
+                - name
+                - value
+              properties:
+                name:
+                  description: name identifies a feature.
+                  type: string
+                value:
+                  description: value is a boolean that specifies whether the feature is activated or not.
+                  type: boolean
+~~~
+
+
+> **Note**
+> disableCopiedCSVs is an existing feature and not part of this enhancement proposal. It predates this enhancement proposal and gets further discussed under [Implementation Details](#DisableCopiedCSVs).
+
+Here is a configuration example:
+
+~~~
+spec:
+  features:
+    disableCopiedCSVs: true
+    activation/gates
+      - name: disableCopiedCSVs
+        value: true
+      - name: enableOptionalManifest
+        value: true
+~~~
+
+### User Stories
+
+#### Stories: Authoring
+
+- As an author I want to be able to set a feature behind a gate without requiring any API change and minimal code changes.
+
+- As an author I want to get feedback on my feature implementation before it has fully matured.
+
+- As an author I want to get feedback on my feature implementation before I commit to its stability in API and behaviour.
+
+- As an author I want to clearly set the expecations in terms of feature maturity and don't want features that are not mature to be activated by default.
+
+#### Stories: Consuming
+
+- As a user I want to understand the product development direction and its impact on my use of it hands-on.
+
+- As a cluster operator I want to avoid features that could lower the stability or consistency of my production clusters.
+
+- As a cluster operator I don't want to rely on features that are not yet mature.
+
+- As a cluster operator I want to be able to quickly deactivate a feature when it has been identified that it lowers the stability or consistency of my clusters.
+
+### Implementation Details
+
+#### OLMConfig
+
+A new version v2alpha1 of the OLMConfig API will be created and supported along v1. This version will introduce a list named `gates` for activation and deactivation of features:
```

Comment:
Should we add a field to the `olmConfig.status` that communicates the status of the enabled/disabled feature?

Reply:
An immediate instance where this may be useful is handling the legacy disableCopiedCSVs feature, in which conflicting values between the old and new api can result in an error.